lib/crypto-primitives/src/Cryptography/Cipher/AES256CBC.hs

```diff
@@ -130,6 +128,9 @@ encrypt mode key iv saltM msg = do
            (\c -> cbcEncrypt c initedIV msg') (initCipher key)
  where
    addSalt salt = saltPrefix <> salt
+   padPKCS7 payload
+       | BS.null payload = Nothing
+       | otherwise = Just (PKCS7.pad payload)
```

Comment:
This helper function was added to preserve the existing behaviour of `encrypt`, which rejects zero-length payloads.